microsoft_cve_rag/application/etl/transformer.py

```
@@ -892,6 +1111,51 @@ def transform_msrc_posts(msrc_posts: List[Dict[str, Any]], process_all: bool = F
+
+        # Add non-prefixed properties
+        nvd_properties.extend([
+            "nvd_description", "nvd_published_date",
+            "cwe_id", "cwe_name", "cwe_source", "cwe_url"
+        ])
+
+        # Extract NVD properties for all records (both preprocessed and new)
+        if not preprocessed_records.empty:
+            for nvd_prop in nvd_properties:
+                preprocessed_records[nvd_prop] = preprocessed_records['metadata'].apply(
+                    lambda x: _extract_nvd_properties(x).get(nvd_prop)
```

Ruff (0.8.2)

Function definition does not bind loop variable `nvd_prop`

(B023)